wal/wal.go

```diff
@@ -348,6 +348,12 @@ func (w *WAL) ReleaseLockTo(index uint64) error {
 		}
 	}
 
+	// if no lock index is greater than the release index, we can
+	// release lock upto the last one(excluding).
+	if smaller == 0 && len(w.locks) != 0 {
```

Comment:
1. holding lock 1,2,3,4,5
2. release lock to 2
3. smaller is set to 0
4. smaller is set to len(w.locks) - 1